### DISCUSSION TITLE
How do I create a shortcut for the Night light on Windows 11?

### DISCUSSION THREAD

**Discussion by gocat** [seerge/g-helper#1847] [Q&A]
I want to create a button（Fn+F4） that toggle night light mode（on/off）, does anyone know how? 
thanks

**Comment by Kikimahe** [comment]
Hi, 

I've searched.

This is not possible at the moment.
In Windows there is no shortcut to switch directly to night mode.

I analyzed the changes in the windows registry with day mode and night mode and didn't find a key that could have easily enabled/disabled.

On Google, many people have searched for a solution, without success. 
This feature was requested from Microsoft, so maybe one day it will come with a Microsoft update... :)

**Comment by aouronga** [comment]
Hi @gocat and @Kikimahe, I created a .bat script to add a shortcut for Windows 11 Night Light. You can follow: https://github.com/aouronga/windows11-night-light-keyboard-shortcut

**Reply by gocat** [reply]
Great work! It works flawlessly!

**Reply by kreatinbiceps** [reply]
Thanks @aouronga !

**Comment by zroxlol** [comment]
I used aouronga's as a template and made one that checks the existing setting, I found that when running the OG script it was not toggling it off when it was already on, this one checks the state of the setting and then if it is on it will press Shift + Tab then Enter and if its off it will only press enter.
It works more routinely than I was experiencing, however if you continuously run it there becomes a strange lag..

I'm still looking into a non scripted method, while this solution works it does feel clunky to me. (Which is Microsofts fault)

@echo off
start ms-settings:nightlight

REM If waiting is necessary, simply uncomment the line below (remove 'REM').
timeout /t 1 /nobreak >nul

REM Check if Night Light is enabled
powershell -command "$nlStatus = Get-ItemProperty -Path 'HKCU:\Software\Microsoft\Windows\CurrentVersion\CloudStore\Store\Cache\DefaultAccount\*windows.data.bluelightreduction.bluelightreductionstate*' | Select-String -Pattern '\"Data\":\[1\]'; if ($nlStatus) { exit 1 } else { exit 0 }"
if %errorlevel%==1 (
    REM Night Light is ON, needs Shift+Tab before Enter
    powershell -command "$wshell = New-Object -ComObject wscript.shell; $wshell.SendKeys('+{TAB}')"
    timeout /t 1 /nobreak >nul
)

REM Press Enter to toggle Night Light
powershell -command "$wshell = New-Object -ComObject wscript.shell; $wshell.SendKeys('{ENTER}')"

REM If waiting is necessary, simply uncomment the line below (remove 'REM').
timeout /t 1 /nobreak >nul

REM Close the Settings window
powershell -command "Get-Process | Where-Object { $_.MainWindowTitle -like 'Settings' } | ForEach-Object { $_.CloseMainWindow() }"

**Reply by SparkyKestrel** [reply]
@zroxlol I needed something like your script (thanks too to @Aouronga for the original idea) as the Nightlight settings on my new PC running W11 Pro 24H2 (preinstalled) will work if I choose "sunset to sunrise" but refuse to work if I set my own on/off times.

I copied your script and ran it, it works great, but is that last line supposed to close the settings window? Because it isn't for me (both turning on and off).
I tried increasing the timeout to wait 10 seconds before executing the close command to see if that would work but the settings window still stays open.

A non scripted method would be nice too if you can work out how to do it.

**Reply by Enyium** [reply]
I wrote a [command line program](https://github.com/Enyium/sem-reg-rs?tab=readme-ov-file#night-light-command-line-program) some time ago that allows you to control Night Light.
